modules/microservice/main.tf

```diff
@@ -50,11 +50,11 @@ locals {
   # 24 characters is used for max key vault and storage account names
   max_name_length                      = 24
 
-  max_environment_differentiator_short = local.max_name_length - (length(var.name) + length(var.environment) + 2)
+  max_environment_differentiator_short = local.max_name_length - (length(var.service_name) + length(var.name) + length(var.environment) + 2)
```

Comment:
Could you change this to be a configurable preference in the root module? This way we can keep these changes for new implementations that want to use it, without breaking the existing one.